editor/resources/prelude.lua

```diff
@@ -99,4 +101,282 @@ editor.ui.resource_field = editor.ui.component(function(props)
+        text = text,
+        issue = error and {severity = editor.ui.ISSUE_SEVERITY.ERROR, message = error}
+    })
+end
+
+function editor.bundle.external_file_field(config, set_config, key, error, rest_props)
+    local props = rest_props or {}
+    props.value = config[key]
+    props.on_value_changed = function(value) set_config(editor.bundle.assoc, key, value or "") end
+    props.issue = error and {severity = editor.ui.ISSUE_SEVERITY.ERROR, message = error}
+    return editor.ui.external_file_field(props)
+end
+
+function editor.bundle.dialog(heading, config, hint, error, content)
+    return editor.ui.dialog({
+        title = "Bundle Application",
+        header = editor.ui.vertical({children = {
+            editor.ui.heading({text = heading}),
+            editor.ui.paragraph({
+                text = error or hint or "Proceed to select output folder.",
+                color = error and editor.ui.COLOR.ERROR or editor.ui.COLOR.HINT
+            })
+        }}),
+        content = editor.ui.grid({
+            padding = editor.ui.PADDING.LARGE,
+            columns = {{}, {grow = true}},
+            children = content
+        }),
+        buttons = {
+            editor.ui.dialog_button({
+                text = "Close",
+                cancel = true
+            }),
+            editor.ui.dialog_button({
+                text = "Create Bundle...",
+                default = true,
+                result = config,
+                enabled = not error
+            }),
+        }
+    })
+end
+
+-- Bundling: flow
+
+function editor.bundle.config(requested_dialog, prefs_key, dialog_component, errors_fn)
+    -- TODO: remove migration from legacy prefs after sufficient time has passed (e.g. after 2026-01-13)
+    if not editor.prefs.is_set(prefs_key .. ".variant") and editor.prefs.is_set("bundle.variant") then
+        pcall(editor.prefs.set, prefs_key .. ".variant", editor.prefs.get("bundle.variant"))
+        editor.prefs.set(prefs_key .. ".texture_compression", editor.prefs.get("bundle.texture-compression"))
+        editor.prefs.set(prefs_key .. ".with_symbols", editor.prefs.get("bundle.debug-symbols"))
+        editor.prefs.set(prefs_key .. ".build_report", editor.prefs.get("bundle.build-report"))
+        editor.prefs.set(prefs_key .. ".liveupdate", editor.prefs.get("bundle.liveupdate"))
+        editor.prefs.set(prefs_key .. ".contentless", editor.prefs.get("bundle.contentless"))
+    end
+    local config = editor.prefs.get(prefs_key)
+    local errors = errors_fn and errors_fn(config)
+    local show_dialog = requested_dialog or errors
+    if show_dialog then
+        config = editor.ui.show_dialog(dialog_component({config = config, errors = errors}))
+        if not config then
+            error(editor.bundle.abort_message)
+        end
+        editor.prefs.set(prefs_key, config)
+    end
+    return config
+end
+
+function editor.bundle.output_directory(requested_dialog, output_subdir)
```

Comment:
The "get output directory" step. Translated from now-deleted `editor.bundle-dialog` ns.